### PR TITLE
fix nri removeMount does not work bug

### PR DESCRIPTION
**Pull request by liangjingtao11**
nri needs to pass the del mount info to containerd, so that containerd can handle the mount records that need to be deleted correctly

fix: https://github.com/containerd/nri/issues/80

### CODE REVIEW

**Comment by mikebrow** [comment]
https://github.com/containerd/nri/blob/main/pkg/api/adjustment.go#L20-L32 for consideration...

File: pkg/adaptation/result.go [hunk 1]
```diff
@@ -322,6 +322,11 @@ func (r *result) adjustMounts(mounts []*Mount, plugin string) error {
 		r.reply.adjust.Mounts = append(r.reply.adjust.Mounts, m)
 	}
 
+	// need add del mount, containerd needs to process this del mounts
```
Comment (klihub) [inline pkg/adaptation/result.go:325]: nitpick: I would change 'containerd' in the above comment to 'runtime'. Maybe rephrase to something like ' add deleted mounts, the runtime needs them to properly remove deleted mounts'.

**Comment by klihub** [comment]
@liangjingtao11 It would be nice to have a test case for this.

**Comment by mikebrow** [comment]
@liangjingtao11 do you have time to finish this PR?

File: pkg/adaptation/result.go [hunk 2]
```diff
@@ -322,6 +322,11 @@ func (r *result) adjustMounts(mounts []*Mount, plugin string) error {
...
+	for _, m := range del {
+		r.reply.adjust.Mounts = append(r.reply.adjust.Mounts, m)
+	}
```
Comment (klihub) [inline pkg/adaptation/result.go:328]: @liangjingtao11 I think doing it this way breaks cases where a single plugin does a RemoveMount() followed by an AddMount(), both using the same destination. This is the idiomatic way for a plugin to change a mount in the case where that mount might have been injected (or updated) by another plugin earlier in the chain. With your changes, that will end up with a final adjustment which will cause the runtime to remove such a mount from the container.

**Comment by klihub** [comment]
@liangjingtao11 This fix is not sufficient in it's current form as it breaks replacing a mount with another one to the same destination. I think we'd need a bit of additional code to prevent that from happening. [Maybe something like this](https://github.com/klihub/nri/tree/fixes/mount-removal-adjustment), where 8f509e7d0f2e4e9af7f7acb77e9c18f2e882efaf is an updated version of your commit, and 7ea0a47f00f853b4e970b980b1a24b082c0930f0 adds a new test case for mount removal.

**Comment by klihub** [comment]
@liangjingtao11 Ping. Do you have time to address the remaining issues ?

**Comment by klihub** [comment]
Closing in favor of updated PR #107.
